### PR TITLE
[patch] ✨ Add Stackdriver Monitoring, Tracing and Profiler support

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

### Description:

- Add Stackdriver Monitoring, Tracing and Profiler support to internal/observability package.
- Updated detecting internal changes CI.
- Add a filter for app_version_info metric labels, because Stackdriver Monitoring accepts only less than 10 labels.

### Related Issue:

nothing.

### How Has This Been Tested?:

nothing.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [X] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/observability/exporter/jaeger/jaeger.go [hunk 1]
```diff
@@ -29,7 +29,6 @@ type jaegerOptions = jaeger.Options
 type Jaeger interface {
 	Start(ctx context.Context) error
 	Stop(ctx context.Context)
```
Comment (kpango) [inline internal/observability/exporter/jaeger/jaeger.go:31]: import root interface package and change below
```suggestion
	exporter.Exporter
```

File: internal/observability/exporter/prometheus/prometheus.go [hunk 2]
```diff
@@ -38,7 +38,6 @@ type prometheusOptions struct {
 type Prometheus interface {
 	Start(ctx context.Context) error
 	Stop(ctx context.Context)
```
Comment (kpango) [inline internal/observability/exporter/prometheus/prometheus.go:40]: same

File: internal/observability/exporter/stackdriver/stackdriver.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"context"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"go.opencensus.io/trace"
+)
+
+type Stackdriver interface {
+	Start(ctx context.Context) error
+	Stop(ctx context.Context)
```
Comment (kpango) [inline internal/observability/exporter/stackdriver/stackdriver.go:29]: same

File: internal/observability/observability.go [hunk 4]
```diff
@@ -133,15 +212,15 @@ func (o *observability) PreStart(ctx context.Context) (err error) {
 		return err
 	}
 
-	if o.prometheus != nil {
-		err = o.prometheus.Start(ctx)
+	for _, ex := range o.exporters {
+		err = ex.Start(ctx)
```
Comment (kpango) [inline internal/observability/observability.go:216]: I think if err returns from Start, we should stop previous exporters,

File: internal/observability/observability.go [hunk 5]
```diff
@@ -133,15 +212,15 @@ func (o *observability) PreStart(ctx context.Context) (err error) {
...
 		if err != nil {
 			return err
 		}
 	}
 
-	if o.jaeger != nil {
-		err = o.jaeger.Start(ctx)
+	for _, prof := range o.profilers {
+		err = prof.Start(ctx)
```
Comment (kpango) [inline internal/observability/observability.go:223]: same

**Comment by rinx** [comment]
@kpango Fixed. Please check them.

/rebase

**Pull request merged**